asf_search/ASFProduct.py

```diff
@@ -44,7 +45,17 @@ def download(self, path: str, filename: str = None, session: ASFSession = None,
         """
         if filename is None:
             default_filename = self.properties['fileName']
-        
+        elif filename is not None:
```

Comment:
I think this can just be an `else`? Since it's a negative of the one above it
And you can simplify it one step farther then:

```python
default_filename = self.properties['fileName']
if filename is not None:
    default_filename = "override with new file name"
```